zulipterminal/ui.py

```diff
@@ -90,14 +92,18 @@ def streams_view(self) -> Any:
                         count=unread_count,
                     )
             )
-        self.stream_w = StreamsView(streams_btn_list)
-        w = urwid.LineBox(self.stream_w, title="Streams")
+        stream_view = StreamsView(streams_btn_list)
+        self.stream_w.extend(stream_view.log)  # For multiple calls
+        w = urwid.LineBox(stream_view, title=title)
         return w
 
     def left_column_view(self) -> Any:
         left_column_structure = [
             (4, self.menu_view()),
-            self.streams_view(),
+            self.streams_view(stream_list=self.pinned_streams,
```

Comment:
Since every user has only a few pinned streams, this could be :
```
        left_column_structure = [
            (4, self.menu_view()),
            ('weight', 1, self.streams_view(stream_list=self.pinned_streams,
                                            title="Pinned Streams")),
            ('weight', 5, self.streams_view(stream_list=self.unpinned_streams,
                                            title="Other Streams")),
        ]
```
Also, can we not display the `pinned streams` line_box when there are no pinned streams.

Reply:
You can refer to this [link](http://urwid.org/reference/widget.html#urwid.Pile) for more details.

Reply:
Mainly I had got to this stage and was fairly happy with the result in terms of it being moderately effective, leaving fine-tuning for a later iteration.

In addition:
* For relative length, I wanted maximum rows for the pinned streams of the number of pinned streams, but optimizing heavily for displaying those over the other streams - I assume that's the intent of the `weight` parameter
* I also planned to leave the no-pinned case for later, though I had considered resolving it.

Reply:
Fixing rows for any widget could backfire unless you take into account the height of the terminal (which could get a lot messier). So, I would suggest following the `weight` approach I mentioned.

I believe for no-pinned case a simple if statement will do for now.
Thanks neil.

Reply:
I wasn't looking to fix the number of rows explicitly, but more set a maximum, which I saw in the documentation but hadn't explored enough to implement at this point. So something like the weighting you suggest (prioritizing pinned), but never showing *more* than the number of streams there (I get that at the moment - there are gaps below my last pinned stream)

Reply:
Yeah, that sounds like a better idea.